pkg/block/block.go

```diff
@@ -153,6 +155,14 @@ func upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir st
 		return cleanUp(logger, bkt, id, errors.Wrap(err, "upload index"))
 	}
 
+	// level 1 blocks should not have partition info file
+	if meta.Compaction.Level > 1 {
+		if err := objstore.UploadFile(ctx, logger, bkt, filepath.Join(bdir, PartitionInfoFilename), path.Join(id.String(), PartitionInfoFilename)); err != nil {
```

Comment:
I feel it is kind of weird to have this additional logic in Thanos. Can we just upload the partition file separately in Cortex?

Reply:
We can have discussion on the previous comment.